app/src/main/java/vn/sunasterisk/music_70/data/remote/TrackAsyncTask.kt

```diff
@@ -0,0 +1,52 @@
+package vn.sunasterisk.music_70.data.remote
+
+import org.json.JSONObject
+import vn.sunasterisk.music_70.base.BaseAsyncTask
+import vn.sunasterisk.music_70.data.model.Track
+import vn.sunasterisk.music_70.util.StringUtils
+
+class TrackAsyncTask(val callback: TrackDataSource.TrackCallback): BaseAsyncTask<Track>(){
+    override fun convertJsonToObject(response: String): List<Track> {
+        val tracks = arrayListOf<Track>()
+        try {
+            val objectRoot = JSONObject(response)
+            val collections = objectRoot.getJSONArray(TrackAttributes.COLLECTION)
+
+            for (i in 0 until collections.length()) {
+                val trackList = collections.getJSONObject(i)
+                val trackObject = trackList.getJSONObject(TrackAttributes.TRACK)
+                val id = trackObject.getInt(TrackAttributes.ID)
+                val title = trackObject.getString(TrackAttributes.TITLE)
+                val	artworkUrl = trackObject.getString(TrackAttributes.ARTWORK_URL)
+                val duration = trackObject.getInt(TrackAttributes.DURATION)
+                val downloadable = trackObject.getBoolean(TrackAttributes.DOWNLOADABLE)
+                val downloadUrl = trackObject.getString(TrackAttributes.DOWNLOAD_URL)
+                val streamUrl = StringUtils.generateStreamUrl(id)
+                val username = trackObject.getJSONObject(TrackAttributes.USER).getString(TrackAttributes.USERNAME)
+                var artist = ""
+                if (!trackObject.isNull(TrackAttributes.PUBLISHER_METADATA)
+                    && !trackObject.getJSONObject(TrackAttributes.PUBLISHER_METADATA)
+                        .isNull(TrackAttributes.ARTIST)) {
+                    artist = trackObject.getJSONObject(TrackAttributes.PUBLISHER_METADATA)
+                        .getString(TrackAttributes.ARTIST)
+                } else {
+                    artist = username
+                }
+                val likeCount= trackObject.getInt(TrackAttributes.LIKE_COUNT)
+                val description = trackObject.getString(TrackAttributes.DESCRIPTION)
+                val track =
+                    Track(id, title, artworkUrl, duration, downloadable, downloadUrl, streamUrl, username, artist,likeCount,description)
```

Comment:
Viết một hàm constructor trong class Track với tham số truyền vào là JSONObject thay vì get dữ liệu ở bên trên thì sẽ gọn hơn nhiều 
```suggestion
                    val track = Track(trackObject)
```
ví dụ như này 
```suggestion
       constructor(jsonObject: JSONObject) : this(
            id = jsonObject.getString(ID),
            title = jsonObject.getString(TITLE),
            images = Image(jsonObject.getJSONObject(IMAGES))
        )
```